### PR TITLE
Me: Create scaffold for me/domains page

### PR DESCRIPTION
**Pull request by jeyip**
Related to https://github.com/Automattic/martech/issues/1872

## Proposed Changes

* Adds the domains management all domains page to me/domains
* Note that if we were to implement this, then we'd be replacing the "Manage All Domains" link in the /me sidebar

## Testing Instructions

* Check out this branch
* `yarn && yarn start`
* Navigate to calypso.localhost:3000/me/domains

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/me/sidebar/index.jsx [hunk 1]
```diff
@@ -114,6 +114,15 @@ class MeSidebar extends Component {
 							preloadSectionName="security"
 						/>
 
+						<SidebarItem
+							selected={ itemLinkMatches( '/domains', path ) }
+							link="/me/domains"
+							label={ translate( 'Domains' ) }
+							materialIcon="settings"
```
Comment (p-jackson) [inline client/me/sidebar/index.jsx:121]: We've been using this globe item everywhere to represent all domains.

```suggestion
							materialIcon="language"
```

**Review by p-jackson — COMMENTED** [review]
Nice, I'm impressed how well it just works out of the box like this.

I'm not sure personally whether putting domains under `/me` is the right thing. The other things in this "domain page emphasis" project are about keeping domains where they are in the hierarchy and just having more links that jump to it. However this seems to change where domains live. And I domains feels like it's own top level thing, not something under accounts.

I know this PR is still in draft, but are you planning on updating each of the links in the domain page so that they link to a new path under `/me/domains/...`?

**Comment by mpkelly** [comment]
I think it feels a bit random to make domains part of the`/me` namespace when other content, like blogs and newsletters, is external. The existing external link "Manage All Domains" seems to be more consistent with what we do for blogs.

**Comment by jeyip** [comment]
@mpkelly @p-jackson 

Hey y'all -- my mistake. Yesterday I left a more detailed discussion of what you both are describing in the issue itself, but I forgot to make a note in this PR as well 😅

https://github.com/Automattic/martech/issues/1872#issuecomment-1613978809

> I'm not sure personally whether putting domains under /me is the right thing. The other things in this "domain page emphasis" project are about keeping domains where they are in the hierarchy and just having more links that jump to it. However this seems to change where domains live. And I domains feels like it's own top level thing, not something under accounts.

> I think it feels a bit random to make domains part of the/me namespace when other content, like blogs and newsletters, is external. The existing external link "Manage All Domains" seems to be more consistent with what we do for blogs.

I had the same concerns. Luis appears to agree as well. At the end of the day, I don't think this implementation is worth the effort given all of the nested links that exist within domains/manage. I agree with y'all in feeling like the "Manage All Domains" link is a good solution

> I know this PR is still in draft, but are you planning on updating each of the links in the domain page so that they link to a new path under /me/domains/...?

I posed this question as a technical consideration in https://github.com/Automattic/martech/issues/1872#issuecomment-1613978809. We might get away with not doing so for the time being, but the navigation experience for users would be confusing as they're taken out of the context of /me and dropped into /domains/manage

**Comment by jeyip** [comment]
It seems like the general consensus is to stick with the "Manage All Domains" link in the sidebar for now. I'll close this PR. We can always return to this technical spike if we change our minds -- it's easy enough to re-implement :D

**Pull request closed without merging**